### PR TITLE
HDDS-2886. parse and dump datanode segment file to pritable text

### PR DESCRIPTION
**Pull request by mukul1987**
## What changes were proposed in this pull request?

This change adds a tool to parse datanode's ratis log files, this tool will be useful to debug issues.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-2886

## How was this patch tested?

Unit tests, added the log files into the jira.

### CODE REVIEW

File: hadoop-hdds/tools/src/main/java/org/apache/hadoop/hdds/datanode/cli/ParseDnRatisLogSegment.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    name = "dnRatisparser", mixinStandardHelpOptions = true)
+public class ParseDnRatisLogSegment implements Runnable {
+  @CommandLine.Option(names = {"-s", "--segmentPath"}, required = true,
+      description = "Path of the segment file")
+  private static File segmentFile;
+
+  private static String smToContainerLogString(
+      StateMachineLogEntryProto logEntryProto) {
+    return ContainerStateMachine.
+        smProtoToString(RaftGroupId.randomId(), null, logEntryProto);
```
Comment (bharatviswa504) [inline hadoop-hdds/tools/src/main/java/org/apache/hadoop/hdds/datanode/cli/ParseDnRatisLogSegment.java]: Passing randomGroupID and setting this in pipelineID for ContainerCommandRequestProto. So, when it is printing pipelineID will be dummy value right?

Comment (mukul1987) [reply hadoop-hdds/tools/src/main/java/org/apache/hadoop/hdds/datanode/cli/ParseDnRatisLogSegment.java]: yes, I added a dummy pipeline ID for this.

File: hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/segmentparser/DatanodeParser.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+/**
+ * Command line utility to parse and dump a datanode ratis segment file.
+ */
+@CommandLine.Command(
+    name = "datanode",
+    description = "dump datanode segment file",
+    mixinStandardHelpOptions = true,
+    versionProvider = HddsVersionProvider.class)
+public class DatanodeParser extends BaseLogParser implements Callable<Void> {
```
Comment (bharatviswa504) [inline hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/segmentparser/DatanodeParser.java]: Can we name this class as DatanodeRatisLogParser

File: hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/segmentparser/GenericParser.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+/**
+ * Command line utility to parse and dump any generic ratis segment file.
+ */
+@CommandLine.Command(
+    name = "generic",
+    description = "dump generic ratis segment file",
+    mixinStandardHelpOptions = true,
+    versionProvider = HddsVersionProvider.class)
+public class GenericParser extends BaseLogParser implements Callable<Void> {
```
Comment (bharatviswa504) [inline hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/segmentparser/GenericParser.java]: Can we rename this as GenericRatisLogParser?

**Comment by mukul1987** [comment]
Thanks for the review @bharatviswa504 , addressed comments in the next path.

**Pull request merged**